examples/targeting/add_customer_negative_criteria.rb

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright:: Copyright 2017, Google Inc. All Rights Reserved.
+#
+# License:: Licensed under the Apache License, Version 2.0 (the "License");
+#           you may not use this file except in compliance with the License.
+#           You may obtain a copy of the License at
+#
+#           http://www.apache.org/licenses/LICENSE-2.0
+#
+#           Unless required by applicable law or agreed to in writing, software
+#           distributed under the License is distributed on an "AS IS" BASIS,
+#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+#           implied.
+#           See the License for the specific language governing permissions and
+#           limitations under the License.
+#
+# This example adds various types of negative criteria to a customer. These
+# criteria will be applied to all campaigns for the customer.
+#
+#
+require 'optparse'
+require 'google/ads/google_ads'
+
+def add_customer_negative_criteria(customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  tradgedy_criterion = client.resource(:CustomerNegativeCriterion)
+  tradgedy_criterion.content_label = client.resource(:ContentLabelInfo)
+  tradgedy_criterion.content_label.type = :TRAGEDY
+
+  placement_criterion = client.resource(:CustomerNegativeCriterion)
+  placement_criterion.placement = client.resource(:PlacementInfo)
+  placement_criterion.placement.url = client.wrapper.string("http://example.com")
+
+  ops = [
+    {create: tradgedy_criterion},
+    {create: placement_criterion},
+  ]
+
+  customer_negative_criterion_service = client.service(
+    :CustomerNegativeCriterion
+  )
+
+  response = customer_negative_criterion_service.mutate_customer_negative_criteria(customer_id, ops)
+
+  if response.partial_failure_error != nil
```

Comment:
unless response.partial_failure_error.nil?

Reply:
Although I don't think you can get a partial failure response unless you explicitly set partial failure to true in the request, which you don't do here.

Reply:
I'll remove

Reply:
also, re: unless, I prefer to only use it in the single line form: `return x unless y`. If doing a multi line form, I almost always do `if`, even if the conditional is negative.